### ISSUE TITLE
BCF File Load Error

### ISSUE THREAD

**Issue by thor0411** [hyperspy/hyperspy#1751]
I'm trying to load a SEM-EDS hypermap generated from a Hitachi 8230 SEM with a Bruker FlatQuad EDS detector. I'm getting an error when loading the file as below:

```
 hs.load(filename)
Traceback (most recent call last):

  File "<ipython-input-3-ada763a7934e>", line 1, in <module>
    hs.load(filename)

  File "C:\ProgramData\Anaconda3\lib\site-packages\hyperspy\io.py", line 243, in load
    for filename in filenames]

  File "C:\ProgramData\Anaconda3\lib\site-packages\hyperspy\io.py", line 243, in <listcomp>
    for filename in filenames]

  File "C:\ProgramData\Anaconda3\lib\site-packages\hyperspy\io.py", line 286, in load_single_file
    **kwds)

  File "C:\ProgramData\Anaconda3\lib\site-packages\hyperspy\io.py", line 295, in load_with_reader
    **kwds)

  File "C:\ProgramData\Anaconda3\lib\site-packages\hyperspy\io_plugins\bcf.py", line 1189, in file_reader
    obj_bcf = BCF_reader(filename, instrument=instrument)

  File "C:\ProgramData\Anaconda3\lib\site-packages\hyperspy\io_plugins\bcf.py", line 876, in __init__
    self.header = HyperHeader(header_byte_str, instrument=instrument)

  File "C:\ProgramData\Anaconda3\lib\site-packages\hyperspy\io_plugins\bcf.py", line 585, in __init__
    self._set_sum_edx(root)

  File "C:\ProgramData\Anaconda3\lib\site-packages\hyperspy\io_plugins\bcf.py", line 735, in _set_sum_edx
    str(i))[0].ClassInstance)

IndexError: list index out of range

```

I've attached a file that I am getting the error on. Anything you can do to help would be greatly appreciated.
[test.zip](https://github.com/hyperspy/hyperspy/files/1373207/test.zip)

**Comment by thomasaarholt** [comment]
Can reproduce. What version of Bruker Esprit are you using?

**Comment by thomasaarholt** [comment]
@sem-geologist Wanna have a go?

**Comment by sem-geologist** [comment]
@thomasaarholt , yeah
... those hitachi... :p not a firsttime cousing trouble

**Comment by sem-geologist** [comment]
"FlatQuad EDS"... what a nice piece of hardware you have :)

**Comment by thomasaarholt** [comment]
Very cool design. But not the most inspired name.

**Comment by sem-geologist** [comment]
btw, for bcf xml troubleshooting, I have created gist [https://gist.github.com/sem-geologist/0f733b27fb7a64528c7b5b9a726ec8ad](https://gist.github.com/sem-geologist/0f733b27fb7a64528c7b5b9a726ec8ad) which extracts and pretty prints xml

**Comment by thomasaarholt** [comment]
That's neat. Although, how do people normally view xmls?

**Comment by sem-geologist** [comment]
I dont know. BTW Hitachi made really interesting improvement: which makes the reader fail; all bcf I saw till now begins its spectral nodes with 0 denomination like <SpectralData0>, where more than one spectra can be present (I still should find real life case there they are actually used), bcf is designed to contain multiply "slices" and its sum spectras – thus the numeration. Now, this cleaver Hitachi decided that spectras will start at 1 (no 0)

**Comment by sem-geologist** [comment]
oh and binary SpectrumData is also 1 not 0

**Comment by sem-geologist** [comment]
the version which can open the file was PR. It however needs a bit more testing, I have reworked a bit the way to access the maps inside bcf and it is controlled by index keyword (it was already as optional keyword, but thanks to this bug, it got recovered that it was not working as supposed).
so `load('your_incredible_map.bcf')` (index=None, by default) will get the map and spectra with lowest index (be it starts from 0 or from 1 or from 99). in case of few slices it can be selected (not needed for this file as it have only single slice): e.g. `load('file.bcf', index=2)`, and finally I future-proofed enabling `index='all'`, which would try returning all mapping slices (but I never had seen a single one such file, however the presence of indexes suggest that technologically it is possible)

**Comment by thor0411** [comment]
This is generated with 1.9.4.3329

Thanks!

_________________________________________
Matthew A. Thorseth
Associate Research Scientist
Analytical Sciences – Surface and Interface Characterization
The Dow Chemical Company
1897 Building, Office E82, Midland, MI 48667
phone: 989.633.4317  |  email: mathorseth@dow.com
[http://www.dow.com/about/images/olympics/composite_logo.gif]<http://www.dow.com/>

**Comment by ericpre** [comment]
Fixed in #1752 thanks @sem-geologist. @thor0411, please use the "next_minor_release" branch to read the file.
